src/Shared/Microsoft.DotNet.Web.Authentication.Tests/DefaultAuthorizeActionModelConventionTests.cs

```diff
@@ -0,0 +1,254 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.WebEncoders.Testing;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Web.Authentication.Tests
+{
+    public class DefaultAuthorizeActionModelConventionTests
+    {
+        private readonly ITestOutputHelper _output;
+
+        public DefaultAuthorizeActionModelConventionTests(ITestOutputHelper output)
+        {
+            _output = output;
+        }
+
+        [InlineData("no/no", HttpStatusCode.Unauthorized)]
+        [InlineData("no/anonymous", HttpStatusCode.OK, "No:Anonymous:Value")]
+        [InlineData("no/any", HttpStatusCode.Unauthorized)]
+        [InlineData("no/role", HttpStatusCode.Unauthorized)]
+        [InlineData("anonymous/no", HttpStatusCode.OK, "Anonymous:No:Value")]
+        [InlineData("anonymous/anonymous", HttpStatusCode.OK, "Anonymous:Anonymous:Value")]
+        [InlineData("anonymous/any", HttpStatusCode.OK, "Anonymous:Any:Value")]
+        [InlineData("anonymous/role", HttpStatusCode.OK, "Anonymous:Role:Value")]
```

Comment:
I think these 2 might be wrong. If the controller has [AllowAnonymous] on it, but the methods have [Authorize] on them, they should reject the request correct?

Reply:
That doesn't seem to be what ASP.NET does, even without our changes.  It's... weird... but our attribute doesn't change this behavior (which this test enshrines)

Reply:
Maybe this changed when they made it not a filter anymore, so the filter code can't do both things at once and resolve the behavior.